### PR TITLE
feat: [PL-25011]: added error handling for multiple SCM APIs [Github provider type only]

### PR DESCRIPTION
**Pull request by mohitg0795**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32838)

### CODE REVIEW

**Review by ankit-harness — DISMISSED** [review]
:lgtm:

> Reviewed 24 of 24 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32838#-:-N1O9w5p9pymJvVMfW6t:bnfp4nl)* status: :shipit: complete! ✓

**Review by ankit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32838)* status: 43 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/test/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateFileScmApiErrorHandlerTest.java` line 47 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1aiI3rApuEY3Du_lJf:-N1aiI3rApuEY3Du_lJg:bkvrtyl) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/test/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateFileScmApiErrorHandlerTest.java#L47)):*
> ```Java
>   @Owner(developers = MOHIT_GARG)
>   @Category(UnitTests.class)
>   public void testHandleErrorWhenStatusCodeIs401() {
> ```

We should say testHandleErrorOnUnauthorizedResponse
___
*[`136-git-sync-manager/src/test/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateFileScmApiErrorHandlerTest.java` line 51 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1aiEmZA1PgaAyVhOeu:-N1aiEmZA1PgaAyVhOev:bejovzb) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/test/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateFileScmApiErrorHandlerTest.java#L51)):*
> ```Java
>       githubCreateFileScmApiErrorHandler.handleError(401, errorMessage);
>     } catch (Exception ex) {
>       WingsException exception = ExceptionUtils.cause(ScmUnauthorizedException.class, ex);
> ```

What's this doing? Its not clear

**Review by bhavya181 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32838)* status: 43 file(s) to review. 2 change requests. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateBranchScmApiErrorHandler.java` line 25 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1bdYbD5K96bQ9ldQ0b:-N1bdYbD5K96bQ9ldQ0c:bp791ip) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateBranchScmApiErrorHandler.java#L25)):*
> ```Java
> public class GithubCreateBranchScmApiErrorHandler implements ScmApiErrorHandler {
>   public static final String GET_FILE_WITH_INVALID_CREDS =
>       "Couldn't fetch requested file from Github as the credentials provided in the connector are invalid or have expired.";
> ```

Couldn't fetch requested file from Github. Credentials provided in the connector are invalid or have expired

**Review by bhavya181 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32838)* status: 43 file(s) to review. 2 change requests. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateFileScmApiErrorHandler.java` line 26 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1bdzzWAl9ddLOiMJ6e:-N1bdzzWAl9ddLOiMJ6f:bb7od0r) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateFileScmApiErrorHandler.java#L26)):*
> ```Java
> public class GithubCreateFileScmApiErrorHandler implements ScmApiErrorHandler {
>   public static final String CREATE_FILE_WITH_INVALID_CREDS =
>       "Couldn't create file in Github as the credentials provided in the connector are invalid or have expired.";
> ```

same
___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreatePullRequestScmApiErrorHandler.java` line 26 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1be2-71zO6G9LCGQJM:-N1be2-71zO6G9LCGQJN:bl5f0hh) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreatePullRequestScmApiErrorHandler.java#L26)):*
> ```Java
> public class GithubCreatePullRequestScmApiErrorHandler implements ScmApiErrorHandler {
>   public static final String CREATE_PULL_REQUEST_WITH_INVALID_CREDS =
>       "We couldn't create pull request in Github as the credentials provided in the connector are invalid or have expired.";
> ```

here also msg is different, we should be consistent everywhere, and why are we not using common constant methods and putting these strings here?

**Review by mohitg0795 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32838)* status: 43 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateBranchScmApiErrorHandler.java` line 25 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1bdYbD5K96bQ9ldQ0b:-N1beM8RBZDoYA2DIOOR:btyg8ku) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateBranchScmApiErrorHandler.java#L25)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

Couldn't fetch requested file from Github. Credentials provided in the connector are invalid or have expired
</blockquote></details>

Any concerns here? Earlier it wasn't grammatically correct, so fixed it.
___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateFileScmApiErrorHandler.java` line 26 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1bdzzWAl9ddLOiMJ6e:-N1beR7w--q_i5wOhqIx:bb7od0r) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateFileScmApiErrorHandler.java#L26)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

same
</blockquote></details>

same
___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreatePullRequestScmApiErrorHandler.java` line 26 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1be2-71zO6G9LCGQJM:-N1beT4C6rN-tN0zl6zN:bn8cy4k) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreatePullRequestScmApiErrorHandler.java#L26)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

here also msg is different, we should be consistent everywhere, and why are we not using common constant methods and putting these strings here?
</blockquote></details>

Message is same, whats the difference? Action item is different, reason is same.
___
*[`136-git-sync-manager/src/test/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateFileScmApiErrorHandlerTest.java` line 47 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1aiI3rApuEY3Du_lJf:-N1be_jrAY-vihu0U50v:bc02se0) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/test/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateFileScmApiErrorHandlerTest.java#L47)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

We should say testHandleErrorOnUnauthorizedResponse
</blockquote></details>

Sure, will fix it.
___
*[`136-git-sync-manager/src/test/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateFileScmApiErrorHandlerTest.java` line 51 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1aiEmZA1PgaAyVhOeu:-N1bebC67CJ_S9-geRBC:b-h7tlgf) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/test/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateFileScmApiErrorHandlerTest.java#L51)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

What's this doing? Its not clear
</blockquote></details>

So, the check is basically to see if we are throwing expected scm exception on different status code. Thus, we catch the exception and then check if expected scm exception is there or not. As its a nested exception, thus we are using util here.

**Review by bhavya181 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32838)* status: 43 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateBranchScmApiErrorHandler.java` line 25 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1bdYbD5K96bQ9ldQ0b:-N1bfIdJBpgywBMZiCEm:butuzrw) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateBranchScmApiErrorHandler.java#L25)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

Any concerns here? Earlier it wasn't grammatically correct, so fixed it.
</blockquote></details>

Yes, @ankit-harness  told to show the message and like this and in listBranch and listRepo error handlers, this is what is integrated
___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreatePullRequestScmApiErrorHandler.java` line 26 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1be2-71zO6G9LCGQJM:-N1bfZQP3Po06m9Lx4EJ:b-36rcsh) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreatePullRequestScmApiErrorHandler.java#L26)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

Message is same, whats the difference? Action item is different, reason is same.
</blockquote></details>

Message is not same, here, "We" is used, which again @ankit-harness  told not to use "we" in error messages and this has been present at many places, please remove it

**Comment by mohitg0795** [comment]
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreatePullRequestScmApiErrorHandler.java` line 26 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1be2-71zO6G9LCGQJM:-N1bgLR-6Bm4Ro6AA04s:b-iqt5qo) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreatePullRequestScmApiErrorHandler.java#L26)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

Message is not same, here, "We" is used, which again @ankit-harness  told not to use "we" in error messages and this has been present at many places, please remove it
</blockquote></details>

Oh ok, typo mistake while copying, will fix it.

**Comment by mohitg0795** [comment]
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateBranchScmApiErrorHandler.java` line 25 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1bdYbD5K96bQ9ldQ0b:-N1bgS117GQCxQ4A1On1:bzc5mnr) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateBranchScmApiErrorHandler.java#L25)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

Yes, @ankit-harness  told to show the message and like this and in listBranch and listRepo error handlers, this is what is integrated
</blockquote></details>

I don't understand why would we want to show 2 lines which isn't also grammatically correct way to show it. I will check with @ankit-harness if there are any concerns. I have a common meeting with him in sometime. We can get this one resolved, shouldn't be a concern.

**Comment by mohitg0795** [comment]
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateBranchScmApiErrorHandler.java` line 25 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1bdYbD5K96bQ9ldQ0b:-N1c1cDs94sthYYirtFd:bjcjkhl) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateBranchScmApiErrorHandler.java#L25)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

I don't understand why would we want to show 2 lines which isn't also grammatically correct way to show it. I will check with @ankit-harness if there are any concerns. I have a common meeting with him in sometime. We can get this one resolved, shouldn't be a concern.
</blockquote></details>

As discussed with @ankit-harness , we thought to check with @rashminandasahoo for the final call. She has suggested the following:
"The requested file could not be fetched from Github. The credentials in the connection are either invalid or expired."

Will go ahead with the same for all such messages. Please comment if there are any concerns with it.

**Review by ankit-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32838)* status: 43 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateBranchScmApiErrorHandler.java` line 25 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1bdYbD5K96bQ9ldQ0b:-N1cA1vs2kTxiuBGPcNB:b-7xhq4a) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateBranchScmApiErrorHandler.java#L25)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

As discussed with @ankit-harness , we thought to check with @rashminandasahoo for the final call. She has suggested the following:
"The requested file could not be fetched from Github. The credentials in the connection are either invalid or expired."

Will go ahead with the same for all such messages. Please comment if there are any concerns with it.
</blockquote></details>

Looks good to me. This is very user friendly msg.

**Comment by mohitg0795** [comment]
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateBranchScmApiErrorHandler.java` line 25 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1bdYbD5K96bQ9ldQ0b:-N1ca6yQ-szaGBp_5NU1:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateBranchScmApiErrorHandler.java#L25)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Looks good to me. This is very user friendly msg.
</blockquote></details>

done

**Comment by mohitg0795** [comment]
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateFileScmApiErrorHandler.java` line 26 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1bdzzWAl9ddLOiMJ6e:-N1ca9w6DxGEEgUXdXwA:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateFileScmApiErrorHandler.java#L26)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

same
</blockquote></details>

done

**Comment by mohitg0795** [comment]
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreatePullRequestScmApiErrorHandler.java` line 26 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1be2-71zO6G9LCGQJM:-N1caAwH9FqioTBJkjtB:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreatePullRequestScmApiErrorHandler.java#L26)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

Oh ok, typo mistake while copying, will fix it.
</blockquote></details>

done

**Comment by mohitg0795** [comment]
*[`136-git-sync-manager/src/test/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateFileScmApiErrorHandlerTest.java` line 47 at r2](https://reviewable.io/reviews/harness/harness-core/32838#-N1aiI3rApuEY3Du_lJf:-N1ca3v8Bg_LNG82GDgQ:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/06f8dc0012674cd73238336b4c9634a4959f6463/136-git-sync-manager/src/test/java/io/harness/gitsync/common/scmerrorhandling/handlers/github/GithubCreateFileScmApiErrorHandlerTest.java#L47)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

Sure, will fix it.
</blockquote></details>

done

**Comment by mohitg0795** [comment]
trigger messagecheck

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 31 of 43 files at r2, 12 of 12 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32838#-:-N1flcR7CUC8Qqq8H0yv:bnfp4nl)* status: 57 file(s) to review. 1 change requests. 3. unresolved discussions. 1 of 2 approvals obtained.

**Review by bhavya181 — DISMISSED** [review]
> Reviewed 14 of 24 files at r1, 31 of 43 files at r2, 12 of 12 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32838)* status: :shipit: complete! ✓

File: 952-scm-java-client/src/main/java/io/harness/exception/SCMExceptionExplanations.java [hunk 1]
```diff
@@ -16,4 +16,39 @@ public class SCMExceptionExplanations {
       + "1. Missing permissions to fetch selected branch of the repo\n"
       + "2. Repo does not exist or has been deleted\n"
       + "3. Credentials are invalid or have expired";
+  public static final String LIST_REPO_WITH_INVALID_CREDS =
+      "We couldn't list repositories as the credentials provided in connector are invalid or have expired.";
```
Comment (bhavya181) [inline 952-scm-java-client/src/main/java/io/harness/exception/SCMExceptionExplanations.java:20]: Why these strings are added here and also it is having "we" in it.

Comment (bhavya181) [reply 952-scm-java-client/src/main/java/io/harness/exception/SCMExceptionExplanations.java:20]: ok, so can you create a ticket for this

**Comment by mohitg0795** [comment]
*[`952-scm-java-client/src/main/java/io/harness/exception/SCMExceptionExplanations.java` line 20 at r4](https://reviewable.io/reviews/harness/harness-core/32838#-N1KykGNVh6_qqOWyUDx-r4-20:-N1gq7Uq6Nlzj_KH2uFO:bpzvdwc) ([raw file](https://github.com/harness/harness-core/blob/63fb2fff3da5ab851936d71369dd4fd987bad457/952-scm-java-client/src/main/java/io/harness/exception/SCMExceptionExplanations.java#L20)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

ok, so can you create a ticket for this
</blockquote></details>

already a ticket for it: https://harness.atlassian.net/browse/PL-24966

**Review by bhavya181 — APPROVED** [review]
> Reviewed 24 of 24 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32838)* status: 24 file(s) to review. 1 of 2 approvals obtained.

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 14 of 24 files at r1, 7 of 44 files at r2, 12 of 12 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32838)* status: 24 file(s) to review.

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 1 of 44 files at r2, 24 of 24 files at r4.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32838)* status: :shipit: complete! ✓

**Pull request merged**